CONTRIBUTING.md

```diff
@@ -2,7 +2,61 @@
 
 # Contributing to the Inverter Network Contracts
 
-Thanks for your interest in improving the Inverter Network contracts!
+Thanks for your interest in improving the Inverter Network contracts! Your contributions play a vital role in enhancing the robustness and usability of the network. Open-source projects like ours thrive on community collaboration, and every contribution, big or small, pushes us closer to our shared vision.
+
+## Using GitHub Issues
+GitHub issues are an essential tool for collaboration and tracking in our project. If you're an external contributor and come across something that isn't working as expected or have suggestions, concerns, or any other reasons to reach out:
+
+- **Open a New Issue**: Please create a new issue detailing the problem, enhancement, or suggestion. Be as descriptive as possible, providing steps to reproduce (if applicable), expected, and actual behavior.
+- **Issue Triage by the Team**: Once an issue is opened, our team will tag and prioritize it accordingly. We use various tags to classify and manage issues effectively.
+- **Good First Issues**: If you're new to the project and looking for a place to start, look for issues tagged with "good first issue". These are typically more straightforward tasks that are a great way to familiarize yourself with the codebase and contribute.
+Remember, every issue opened helps improve the project, and your feedback is invaluable to us!
+
+## Branching System
+
+We operate using a two-branch-system:
+- **main**: This is our stable branch. Releases are made from this branch.
+- **dev**: Development and feature integration happens here.
+
+## Workflow
+1. Create feature branches from `dev` when working on any new feature or bug fix.
+2. Once your feature is complete, create a PR to merge from your feature branch into `dev`.
+3. When ready to release to `main`, the team will create a PR from `dev` to `main`.
+4. External contributors are welcome to create PRs that merge into `dev`.
+
+## Criteria for Pull Requests
+
+### Merging into `dev`
+Your PR should have:
+- Full test coverage of the new feature.
```

Comment:
Hm maybe add here something in the lines of:
You can use our existing test files as an orientation. We already created abstract testfiles for modules as well as a lot of mocks.....